File: 11_flutter_push_notifications/server/lib/notification_input_validator.dart
class InputValidationException implements Exception {
  InputValidationException(this.message);

  final String message;

  @override
  String toString() => message;
}

class NotificationInputValidator {
  static const maxTitleLength = 30;
  static const maxBodyLength = 100;

  void validateTitle(String title) {
    final trimmed = title.trim();
    if (trimmed.isEmpty) {
      throw InputValidationException('Title cannot be empty.');
    }
    if (trimmed.length > maxTitleLength) {
      throw InputValidationException(
        'Title must be at most $maxTitleLength characters.',
      );
    }
  }

  void validateBody(String body) {
    final trimmed = body.trim();
    if (trimmed.isEmpty) {
      throw InputValidationException('Body cannot be empty.');
    }
    if (trimmed.length > maxBodyLength) {
      throw InputValidationException(
        'Body must be at most $maxBodyLength characters.',
      );
    }
  }
}

File: 11_push_notification/server-net/src/ServerNet/NotificationInputValidator.cs
namespace ServerNet;

public sealed class NotificationInputValidator
{
    public const int MaxTitleLength = 30;
    public const int MaxBodyLength = 100;

    public void ValidateTitle(string title)
    {
        var trimmed = title.Trim();
        if (trimmed.Length == 0)
        {
            throw new InputValidationException("Title cannot be empty.");
        }

        if (trimmed.Length > MaxTitleLength)
        {
            throw new InputValidationException(
                $"Title must be at most {MaxTitleLength} characters.");
        }
    }

    public void ValidateBody(string body)
    {
        var trimmed = body.Trim();
        if (trimmed.Length == 0)
        {
            throw new InputValidationException("Body cannot be empty.");
        }

        if (trimmed.Length > MaxBodyLength)
        {
            throw new InputValidationException(
                $"Body must be at most {MaxBodyLength} characters.");
        }
    }
}
